src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -1255,8 +1462,8 @@ internal static bool IsSubsetOfHelper(ICollection subset, ICollection superset)
         // $ CONSIDER: comparison, which should result in ~n*log(n) + m*log(m) + n.
 
         // Count the occurrences of each object in both collections.
-        Dictionary<object, int> subsetElements = GetElementCounts(subset, out int subsetNulls);
-        Dictionary<object, int> supersetElements = GetElementCounts(superset, out int supersetNulls);
+        Dictionary<object, int> subsetElements = GetElementCounts(subset.Cast<object>(), EqualityComparer<object>.Default, out int subsetNulls);
```

Comment:
Could we have the 2 collections as `IEnumerable<T>` on the signature or is it having too many impacts?

Reply:
this function is used only inside an API which is not generic what is the point of making it generic?

Reply:
None :)